### ISSUE TITLE
Support Codeberg repositories

### ISSUE THREAD

**Issue by cassidyjames** [elementary/houston#797]
We've gotten some people asking to support Codeberg as a source for repositories. While you can set up a mirror on GitHub (and that's my current recommendation), some are opposed to using GitHub at all.

**Comment by 4jNsY6fCVqZv** [comment]
@cassidyjames Could you imagine adding this issue to the AppCenter For Everyone Sprint?
https://github.com/orgs/elementary/projects/72

**Comment by 4jNsY6fCVqZv** [comment]
Codeberg is based on https://en.wikipedia.org/wiki/Gitea. So a Codeberg integration also means Gitea support, right? Gitea is self-hostable.

**Comment by jalcine** [comment]
This is something I'm adamant about supporting mainly because it would be the in-road to allowing elementary to interact with the broader open source community that attempts to reduce their engagement with organizations like ICE. I'm familiar with the API exposed by Gitea and from what I can gather about Houston's use of GitHub, it can definitely be compatible with it.

**Comment by jalcine** [comment]
I'd ask if elementary, as a team, is open to this because it's not a small undertaking (I can imagine a lot of bits about GitHub being hard-coded into Houston) at all.

**Comment by jalcine** [comment]
Looking at the code in https://github.com/elementary/houston/blob/v2/src/lib/service/github.ts, it doesn't look too impossible!

**Comment by btkostner** [comment]
Sorry about the delay (I get way too many notifications on GitHub). I appreciate the work, and this is something we definitely want to support. But the next iteration of houston (with flatpak support) will most likely be a rewrite of this code base in a different language with some core differences to support other source controls besides GitHub.

**Comment by jalcine** [comment]
Hey @btkostner,

> Sorry about the delay (I get way too many notifications on GitHub).

Heh, I declare bankruptcy on them once a month!

> I appreciate the work, and this is something we definitely want to support. But the next iteration of houston (with flatpak support) will most likely be a rewrite of this code base in a different language with some core differences to support other source controls besides GitHub.

Is it possible for the community to contribute to this? I have a vested interest in this and would have no problem contributing time.

**Comment by btkostner** [comment]
@jalcine The server side of the project will be here https://github.com/elementary/houston-server. We were planning on figuring everything out at the AppCenter for Everyone sprint, but [that was sadly postponed](https://blog.elementary.io/appcenter-for-everyone-remote-sprint/). Luckily Dan, Cassidy, and I are planning a mini sprint all next week to figure out what we need to do for the next iteration of houston. I should have more details for you after that week and hopefully a pretty good place for you to start building out Gitea support.

**Comment by jalcine** [comment]
Excellent, super-excited to hear about that - thank you!

**Comment by jalcine** [comment]
Oh nice, it's in Elixir too! I've enjoyed that language a lot (as well 
as Rust).

**Comment by btkostner** [comment]
Yep! I use it a lot at work at System76 and have fallen in love with it. I'd rather have the backend and web interface use that instead of node. Especially with how fast node moves. Most of the code in this houston repo is super out of date because it was written years ago. Ain't nobody have time for that maintenance work

**Comment by 4jNsY6fCVqZv** [comment]
Should this issue be transferred to https://github.com/elementary/appcenter-dashboard?

**Comment by 4jNsY6fCVqZv** [comment]
@jalcine Could you imagine continuing your work on it?

**Comment by 4jNsY6fCVqZv** [comment]
@jalcine Could you imagine continuing your work on this now that AppCenter Dashboard exists?
